### PR TITLE
DOP-4382: button on /docs/legacy does not link to .tar.gz format

### PR DESCRIPTION
**Pull request by anabellabuckvar**
### Stories/Links:

[DOP-4382](https://jira.mongodb.org/browse/DOP-4382)

### Current Behavior:

[Current](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/DOP-4300/landing/anabella.buckvar/main/legacy/index.html)

### Staging Links:

[staging](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/DOP-4300/landing/anabella.buckvar/DOP-4382/legacy/index.html)

To test, use the `MongoDB Connector for BI` product and select applicable downloadable versions.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Review by schmalliso — APPROVED** [review]
This looks good to me, though I'd like another pair of eyes before we :shipit:

File: src/components/DeprecatedVersionSelector.js [hunk 1]
```diff
@@ -141,6 +141,11 @@ const DeprecatedVersionSelector = () => {
     }
 
     // Utilizing hardcoded env because legacy sites are not available on dev/stage
+    if (currentVersion.icon) {
+      const bucket = `https://docs-mongodb-org-dotcomprd.s3.us-east-2.amazonaws.com/docs/offline`;
+      return `${bucket}/${product}-${currentVersion.value}.tar.gz`;
```
[the remark this answers is not in the document]
Comment (anabellabuckvar) [reply src/components/DeprecatedVersionSelector.js:146]: I updated it- it seems like a lot of the prod data in pool.repos_branches had the 'v' included in selector labels but clearly not all! As such I've updated it to use `urlSlug` instead.

[the remark this answers is not in the document]
Comment (mmeigs) [reply src/components/DeprecatedVersionSelector.js:146]: I could be wrong, but I think this is to ask you to refer to `currentVersion.urlSlug` to have the right path. I believe we shouldn't be changing the `versionSelectorLabels` in Hapley/Atlas. Those strings have wider uses than just this file, and that's more the domain of the content writers.

[the remark this answers is not in the document]
Comment (schmalliso) [reply src/components/DeprecatedVersionSelector.js:146]: IMO `currentVersion.urlSlug` should do the needful, assuming one is set, and if not fall back to `currentVersion.gitBranchName`

[the remark this answers is not in the document]
Comment (schmalliso) [reply src/components/DeprecatedVersionSelector.js:146]: @anabellabuckvar Just to clarify, did you update the PR to use `urlSlug` or the data in Atlas?

[the remark this answers is not in the document]
Comment (anabellabuckvar) [reply src/components/DeprecatedVersionSelector.js]: I updated it to use urlSlug, but turns out I hadn't pushed my changes :o sorry all for the confusion!!!

[the remark this answers is not in the document]
Comment (mmeigs) [reply src/components/DeprecatedVersionSelector.js]: ^^ all good! I misunderstood. No Atlas has been changed :) Anabella and I just talked in the office.

**Review by mmeigs — APPROVED** [review]

**Pull request merged**